python/src/interpreter.cc

```diff
@@ -366,6 +575,12 @@ template <RMWOp Op> struct OpCreator {
     if (!atomic_op && dtype.is(pybind11::dtype::of<T>())) {
       atomic_op = std::make_unique<AtomicRMWOp<T, Op>>(ptr, val, ret, mask,
                                                        numel, order);
+    } else if (!atomic_op &&
+               std::string(pybind11::str(dtype)) == std::string("float16")) {
```

Comment:
The main ugly part because `numpy.float16` is not registered in `pybind11`, and I couldn't find a way to register the type manually.